packages/web3-core/types/index.d.ts

```diff
@@ -0,0 +1,62 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @date 2018
+ */
+
+import {AbstractProviderAdapter, ProvidersModuleFactory, provider} from 'web3-providers';
+
+export class AbstractWeb3Module {
+    constructor(
+        provider: AbstractProviderAdapter | provider | string,
+        providersModuleFactory: ProvidersModuleFactory,
+        // not sure what the below object structure is
+        providers: {},
+        // dont have type yet
+        // as this is in web3-core-method
+        // can be sorted later once dependencies
+        // are cleaned up
+        methodController: any,
```

Comment:
The methodController dependency no longer exists because of the latest improvements.

The new constructor of the AbstractWeb3Module is: https://github.com/ethereum/web3.js/blob/ethereumProvider/packages/web3-core/src/AbstractWeb3Module.js#L26-L43